shared/js/background/storage/tds.es6.js

```diff
@@ -14,17 +15,36 @@ class TDSStorage {
         this.tds = { entities: {}, trackers: {}, domains: {}, cnames: {} }
         this.surrogates = ''
         this.ClickToLoadConfig = {}
-        this.config = extensionConfig
-        this.storeInLocalDB('config', extensionConfig)
+        this.config = {}
+        // If initOnInstall is not called before getList, just resolve.
+        this._installing = Promise.resolve(false)
 
         this.removeLegacyLists()
     }
 
+    initOnInstall () {
+        this._installing = this._internalInitOnInstall()
+    }
+
+    async _internalInitOnInstall () {
+        await settings.ready()
+        const etagKey = 'config-etag'
+        const etagValue = settings.getSetting(etagKey)
+        // If there's an existing value ignore the bundled values
+        if (!etagValue) {
+            settings.updateSetting(etagKey, etags[etagKey])
+            this.config = extensionConfig
+            await this.storeInLocalDB('config', extensionConfig)
+        }
+    }
+
     getLists () {
         return Promise.all(constants.tdsLists.map(list => this.getList(list)))
     }
 
-    getList (list) {
+    async getList (list) {
+        // If initOnInstall was called, await the updating from the local bundles before fetching
+        await this._installing
```

Comment:
I find this pattern rather confusing. We're setting `this._installing` to a promise resolving to false, then to a function call that doesn't return anything (`undefined`), and here we are not calling anything. Isn't there a clearer pattern we can use? Like calling a function that actually changes `_installing` to something meaningful, like `true`/`false` once the promise resolves?

Reply:
It's a weird pattern I agree, we do use it elsewhere for settings for example (it also sets `isReady` which is handy when we can choose to ignore if that it's ready, however in this case we must block to prevent the race).